Second iteration

shruthihsrinath opened the pull request:
Second iteration for Purple Owl Games website :
1. Created about page.
2. JSON data to store the games in detail.
3. A Service module to read the JSON data file.
4. Routing for About page, Games page and Game detail page.
5. Styling for all the web pages.

shruthihsrinath committed:
created about and game detail page and an app router page for routes

shruthihsrinath committed:
Second Iteration - Created Button component and added color palette

shruthihsrinath committed:
modified game details page and created service to read data from json

shruthihsrinath committed:
created Service to read json file and updated router to route and pass data properly to all the pages

shruthihsrinath committed:
restyling the pages and added click functionality for mobile resize

shreeshbhat commented on src/app/games/game-card/game-card.component.html:
```diff
@@ -1,11 +1,9 @@
-<article class="game-card">
-    <header class="game-card-header" alt="game name">{{poBlogData?.gameName}}</header>
+<article class="game-card" (click)="readMore(poBlogData?.gameName)">
```
Use button for click listener

shreeshbhat commented on src/app/games/game-card/game-card.component.ts:
```diff
@@ -13,14 +13,21 @@ export class GameCardComponent implements OnInit {
     this.gameButtonDesc = "More on " + this.poBlogData?.gameName;
     this.gameImageAltText = this.poBlogData?.gameName + " card front";
+
+    if (window.innerWidth == 375 && window.innerHeight == 812)
```
Make this bit generic condition.

shreeshbhat commented on src/app/app.component.css:
```diff
@@ -1,14 +1,14 @@
-    --main-bg-color: #C8BFD4;
-    --main-footer-bg-color: #6E4E8D;
+    --main-bg-color: #e8e5f1;
+    /*--main-bg-color: #C8BFD4;
```
Remove commented code

shreeshbhat commented on src/app/domain/po-blog.model.ts:
```diff
@@ -0,0 +1,7 @@
+export interface PoBlog {
```
You can rename this interface and filename to Blog as it won't be exposed to the user.

shruthihsrinath committed:
Fix second-iteration pull request
Modify GameCardComponent to button for accessibility.
Add observable for window resize.
Rename blog model for better understanding and simplicity.

shreeshbhat commented on src/app/app.component.css:
```diff
@@ -1,40 +1,6 @@
-    .game-cards {
-        min-width: min-content;
-    }
+    font-size: 18px;
```
I like 20px font size better

shreeshbhat commented on src/app/app.module.ts:
```diff
@@ -3,16 +3,32 @@ import { BrowserModule } from '@angular/platform-browser';
-    GameCardComponent
+    GameCardComponent,
+    GamesComponent,
+    AboutComponent,
```
About page components should go to a separate module. 
Make the initial load faster.

shreeshbhat commented on src/app/header/header.component.html:
```diff
@@ -1,9 +1,27 @@
+        <img src="assets/company-name-w.png" alt="Purple Owl Games" class="company-name" />
+    </a>
+    <div class="home">
+        <a routerLink="/home" alt="home page">Home</a>
```
Capitalize alt text

shreeshbhat commented on src/app/header/header.component.html:
```diff
@@ -1,9 +1,27 @@
     </div>
     <div class="about">
-        <a href="#" alt="about">About</a>
+        <a routerLink="/pages/about" alt="about">About</a>
```
Capitalize alt text

shreeshbhat commented on src/app/header/header.component.html:
```diff
@@ -1,9 +1,27 @@
+    <div class="menu-options" *ngIf="displayMenu">
+        <a routerLink="/home" alt="home page">Home</a>
+        <a routerLink="/pages/games" alt="Games List">Games</a>
+        <a routerLink="/pages/about" alt="about">About</a>
```
Capitalize alt text

shreeshbhat commented on src/app/home/home.component.ts:
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-home',
```
Change to po-home to be consistent

shreeshbhat commented on src/app/pages/about/about.component.ts:
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-about',
```
Change to po-about

shreeshbhat commented on src/app/pages/games/game-card/game-card.component.css:
```diff
@@ -0,0 +1,46 @@
+    flex-flow: row;
+    border-radius: inherit;
+    box-shadow: 0 0 16px 0 rgba(73, 73, 73, 1);
+    /*transition: all .3s linear;*/
```
Remove commented code

shreeshbhat commented on src/app/pages/games/game-card/game-card.component.css:
```diff
@@ -0,0 +1,46 @@
+
+.game-card .game-card-header {
+    visibility: hidden;
+    display: none;
```
Is this html content needed?

shreeshbhat commented on src/app/pages/games/game-card/game-card.component.html:
```diff
@@ -0,0 +1,9 @@
+<po-button [altAttribute]="gameButtonDesc" (clickEvent)="readMore(poBlogData?.gameName)" class="game-card">
+    <header class=" game-card-header" alt="game name">{{poBlogData?.gameName}}</header>
+    <img class="game-image" src={{poBlogData?.gameUrl}} alt={{gameImageAltText}} />
+    <div class="game-content" alt="game explanation">
```
Capitalize alt content

shreeshbhat commented on src/app/pages/games/game-card/game-card.component.ts:
```diff
@@ -0,0 +1,36 @@
+import { Blog } from 'src/app/domain/blog.model';
+
+@Component({
+  selector: 'game-card',
```
Change to po-game-card

shreeshbhat commented on src/app/pages/games/game-card/game-card.component.ts:
```diff
@@ -0,0 +1,36 @@
+  gameButtonDesc: string | undefined;
+  gameImageAltText: string | undefined;
+
+  isMobile: boolean | undefined = false;
```
I think you don't need undefined type here cos you have a default boolean value set.

shreeshbhat commented on src/app/pages/games/game-detail/game-detail.component.ts:
```diff
@@ -0,0 +1,31 @@
+import { Blog } from 'src/app/domain/blog.model';
+
+@Component({
+  selector: 'game-detail',
```
Change to po-game-detail

shreeshbhat commented on src/app/pages/games/games.component.ts:
```diff
@@ -0,0 +1,20 @@
+import { Blog } from '../../domain/blog.model';
+
+@Component({
+  selector: 'games',
```
Change to po-games

shruthihsrinath committed:
Fix second iteration pull request
Lazy load pages module
Capitalize all alt text
Prepend all component selectors with po
Remove comments

shreeshbhat commented on src/app/header/header.component.ts:
```diff
@@ -1,10 +1,20 @@
+  menuButton: string = "hamburger menu";
+  displayMenu: boolean = false;
+  constructor() { }
+  menuList(clicked: boolean) {
```
Change this method name to `showMenu()`. `menuList` sounds like you are fetching menus here.

shreeshbhat commented on src/app/header/header.component.ts:
```diff
@@ -1,10 +1,20 @@
+      this.displayMenu = true;
+    }
+    else {
+      this.displayMenu = false;
```
Change boolean variables to have `is` or `has` prefix.

shreeshbhat commented on src/app/services/po-blog-data.service.ts:
```diff
@@ -0,0 +1,17 @@
+@Injectable({
+  providedIn: 'root'
+})
+export class PoBlogDataService {
```
Services need not have Po prefix. Only components need to have them as they are exposed to the DOM.

shreeshbhat reviewed (commented):
Please address the comments and change the commit messages to follow the guidelines listed below : 
https://cbea.ms/git-commit/

shruthihsrinath committed:
Fix lazy loading for pages component and its child components

shruthihsrinath committed:
Resolve comments on pull request
Rename methods appropriately
Decouple css from html
Rename blog service
